stdlib/cache/src/main/ballerina/src/cache/abstract_cache.bal

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents Ballerina `AbstractCache` object and cache related operations. Any custom cache implementation should be
+# object-wise similar.
+public type AbstractCache abstract object {
+
+    # Adds the given key, value pair to the cache.
+    #
+    # + key - Key of the cached value
+    # + value - Value to be cached
+    # + maxAgeInSeconds - The value in seconds which the cache entry is valid. '-1' means, the entry is valid forever.
+    public function put(string key, any value, int maxAgeInSeconds);
+
+    # Returns the cached value associated with the given key. If the provided cache key is not found,
+    # () will be returned.
+    #
+    # + key - Key which is used to retrieve the cached value
+    # + return - The cached value associated with the given key
+    public function get(string key) returns any?;
```

Comment:
`any` contains `()` so `any?` is redundant.

Since a cache can be hold basically any value, this return type could cause the issue of not being able to identify if `()` is returned because the cache does not contain an entry by the particular key vs the cache holding `()` as the value for the particular key.

A similar situation is avoided in the iterator design by having the next method return `record {| T value; |}?`, where returning a record always represents the presence of a value. See https://ballerina.io/spec/lang/2019R3/#section_5.5.1

Reply:
As discussed offline the API is fixed with https://github.com/ballerina-platform/ballerina-lang/pull/21308/commits/26a8a6ee5be15b01996a7ef85476230868369e20. Thanks for the suggestion.